lib/tokenize.cpp

```diff
@@ -9811,13 +9812,18 @@ void Tokenizer::removeUnnecessaryQualification()
     }
 }
 
-void Tokenizer::simplifyDeprecated()
+void Tokenizer::simplifyCPP14Attribute()
 {
-    if (_settings->standards.cpp != Standards::CPP11 || isC())
+    if (_settings->standards.cpp < Standards::CPP11 || isC())
         return; // It is actually a C++14 feature, however, there seems to be nothing dangerous about removing it for C++11 as well
 
     for (Token *tok = list.front(); tok; tok = tok->next()) {
-        if (tok->link() && Token::simpleMatch(tok, "[ [ deprecated")) {
+        if (tok->link() && Token::Match(tok, "[ [ deprecated|noreturn")) {
+            if (tok->strAt(2) == "noreturn") {
+                const Token * head = Token::findsimplematch(tok->tokAt(5), "(");
```

Comment:
This will just lookup the next '('. What do you think about such loop instead?

    while (Token::Match(head, "%name%|::|*|&"))
        head = head->next();
    if (head && head->str() == "(")
        ......

I wonder if the `[[noreturn]]` will always precede the function name or if such attribute is possible:

    void foo() [[noreturn]] {..}